database/migration/src/main/resources/changesets/enlarge_workflow_store_entry_id.xml

```diff
@@ -0,0 +1,82 @@
+                baseTableName="SUB_WORKFLOW_STORE_ENTRY"
+                constraintName="FK_SUB_WORKFLOW_STORE_ENTRY_ROOT_WORKFLOW_ID"
+        />
+    </changeSet>
+
+    <!-- BEGIN WORKFLOW_STORE_ENTRY_ID PK widening -->
+    <!-- For Postgresql there are 2 changesets: one for modifying the table column type, and another one for altering the autoincrementing sequence.
+         The other DBs can be refactored similarly with a single addAutoIncrement changeset. -->
+    <changeSet author="sshah" id="enlarge_workflow_store_entry_id" dbms="mysql,hsqldb,mariadb">
+        <addAutoIncrement
+                columnName="WORKFLOW_STORE_ENTRY_ID"
+                columnDataType="BIGINT"
+                incrementBy="1"
+                tableName="WORKFLOW_STORE_ENTRY"
+        />
+    </changeSet>
+
+    <changeSet author="sshah" id="postgresql_enlarge_workflow_store_entry_id" dbms="postgresql">
+        <modifyDataType
+                columnName="WORKFLOW_STORE_ENTRY_ID"
+                tableName="WORKFLOW_STORE_ENTRY"
+                newDataType="BIGINT"
+        />
+    </changeSet>
+
+    <changeSet author="sshah" id="postgresql_enlarge_workflow_store_entry_id_seq" dbms="postgresql">
+        <preConditions onFail="MARK_RAN">
+            <!-- idempotency check (noop if the sequence is present and already consistent what the alter would do) -->
+            <sqlCheck expectedResult="0">
+                SELECT count(*)
+                FROM information_schema.sequences
+                WHERE sequence_name = 'WORKFLOW_STORE_ENTRY_WORKFLOW_STORE_ENTRY_ID_seq'
+                AND data_type = 'bigint';
+            </sqlCheck>
+        </preConditions>
+        <sql>alter sequence "WORKFLOW_STORE_ENTRY_WORKFLOW_STORE_ENTRY_ID_seq" as bigint;</sql>
+    </changeSet>
+    <!-- END WORKFLOW_STORE_ENTRY_ID PK widening -->
+
+    <!-- BEGIN widening FK to match PK -->
+    <changeSet author="sshah" id="enlarge_sub_workflow_store_entry_fk" dbms="mysql,hsqldb,postgresql,mariadb">
+        <modifyDataType
+                tableName="SUB_WORKFLOW_STORE_ENTRY"
+                columnName="ROOT_WORKFLOW_ID"
+                newDataType="BIGINT"
+        />
+    </changeSet>
+    <!-- END widening FK to match PK -->
+
+    <!-- MariaDB's FK NotNull constraint does not survive the widening above and must be recreated explicitly. -->
+    <!-- BEGIN Restoring FK NotNull constraint -->
+    <changeSet author="sshah" id="mariadb_not_null_constraint_sub_workflow_store_entry_fk" dbms="mariadb,mysql">
+        <addNotNullConstraint
+                tableName="SUB_WORKFLOW_STORE_ENTRY"
+                columnName="ROOT_WORKFLOW_ID"
+                columnDataType="BIGINT"
+        />
+    </changeSet>
+    <!-- END Restoring FK NotNull constraint -->
+
+    <!-- Restoring the FK -->
+    <changeSet author="sshah" id="recreate_sub_workflow_store_entry_root_workflow_id_fk" dbms="mysql,hsqldb,postgresql,mariadb">
+        <addForeignKeyConstraint
```

Comment:
Does the "takes very little time" assumption still hold if we are re-adding indexes as part of this changeset?

Reply:
I am not sure if this was measured on prod clone

Reply:
Do you think it would be better to measure how much time it takes to add the FK constraint back in prod clone before merging this PR?

Reply:
Probably a good idea to double check, just in case it goes from 20ms to 20 hours